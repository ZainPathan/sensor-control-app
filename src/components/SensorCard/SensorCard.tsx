import {
    Button,
    Card,
    CardActions,
    CardContent,
    Grid,
    makeStyles, Typography
} from '@material-ui/core';
import React from 'react';

const useSensorCardStyle = makeStyles({
    button: {
        width: '100%'
    }
});

interface SensorCardProps {
    sensorData: any,
    handleSensorButtonClick: Function
}

const SensorCard: React.FC<SensorCardProps> = ({sensorData, handleSensorButtonClick}) => {
    const classes = useSensorCardStyle();

    return (
        <Grid item xs={12} sm={4} md={4} lg={4}>
            <Card variant="outlined">
                <CardContent>
                    <Typography
                        variant="h5"
                        component="h2"
                    >
                        {sensorData.name}
                    </Typography>
                    <Typography
                        variant="h6"
                        gutterBottom
                    >
                        ID: {sensorData.id}
                    </Typography>
                    <Typography
                        variant="h6"
                        gutterBottom
                    >
                        Connection Status: {sensorData.connected === true ? 'Connected' : 'Disconnected'}
                    </Typography>
                    <Typography
                        variant="h6"
                        gutterBottom
                    >
                        Value(in unit): {sensorData.value ? `${sensorData.value} ${sensorData.unit}` : '-'}
                    </Typography>
                </CardContent>
                <CardActions>
                    <Button
                        variant="contained"
                        color={sensorData.connected === true ? 'secondary' : 'primary'}
                        className={classes.button}
                        onClick={() => {
                            handleSensorButtonClick(sensorData);
                        }}
                        size="large"
                    >
                        {sensorData.connected === true ? 'Disconnect' : 'Connect'}
                    </Button>
                </CardActions>
            </Card>
        </Grid>
    )
}

export default SensorCard;