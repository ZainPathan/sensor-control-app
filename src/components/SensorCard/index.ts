import SensorCard from './SensorCard';

export {
    SensorCard
};

export default SensorCard;