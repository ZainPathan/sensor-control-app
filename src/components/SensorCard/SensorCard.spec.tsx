import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import SensorCard from './SensorCard';

describe('SensorCard tests', () => {

    const mockSensorData = {
        "id": "0",
        "name": "Temperature",
        "connected": true,
        "unit": "Â°C",
        "value": "15.624"
    };
    const mockButtonClick = jest.fn();

    it('snapshot test', () => {
        const tree = renderer
            .create(
                <SensorCard
                    sensorData={mockSensorData}
                    handleSensorButtonClick={mockButtonClick}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('loads with proper Sensor data', () => {
        const mockSensorTitle = 'Temperature';
        const { getByText } = render(
            <SensorCard
                sensorData={mockSensorData}
                handleSensorButtonClick={mockButtonClick}
            />
        );
        expect(getByText(mockSensorTitle)).toBeTruthy();
    });

    it('handler function is called on button click', () => {
        const { getByRole } = render(
            <SensorCard
                sensorData={mockSensorData}
                handleSensorButtonClick={mockButtonClick}
            />
        );
        fireEvent.click(getByRole('button'));
        expect(mockButtonClick).toHaveBeenCalled();
    });
})