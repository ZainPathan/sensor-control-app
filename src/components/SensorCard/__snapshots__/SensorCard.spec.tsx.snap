// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SensorCard tests snapshot test 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
      >
        Temperature
      </h2>
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
      >
        ID: 
        0
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
      >
        Connection Status: 
        Connected
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
      >
        Value(in unit): 
        15.624 Â°C
      </h6>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Disconnect
        </span>
      </button>
    </div>
  </div>
</div>
`;
